model/task_queue_service_dependency.go

```diff
@@ -354,6 +342,27 @@ func (t *basicCachedDAGDispatcherImpl) nextTaskGroupTask(unit schedulableUnit) *
 			return nil
 		}
 
+		// Check if its dependencies have been met.
+		dependencyCaches := make(map[string]task.Task)
+		depsMet, err := nextTaskFromDB.DependenciesMet(dependencyCaches)
+		if err != nil {
+			grip.Warning(message.WrapError(err, message.Fields{
+				"dispatcher": "dependency-task-dispatcher",
+				"function":   "nextTaskGroupTask",
+				"message":    "error checking dependencies for task",
+				"outcome":    "skip and continue",
+				"task":       nextTask.Id,
+			}))
+			continue
+		}
+
+		if !depsMet {
+			// TO DO do we need to set both of theses?
```

Comment:
Could you check this before you merge?

Reply:
Sure - this is still a WIP.

Reply:
Both assignment statements give the same desired affect.